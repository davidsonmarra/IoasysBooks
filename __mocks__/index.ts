import BookDTO from '../src/@types/BookDTO';

interface IMockFetchDataBooks {
  data: BookDTO[];
  page: number;
  totalPages: number;
  totalItems: number;
}

export const mockBook: BookDTO = {
  id: '1',
  title: 'Test Title',
  imageUrl: 'https://www.test.com',
  authors: ['Test Author'],
  pageCount: 100,
  publisher: 'Test Publisher',
  published: 1980,
  category: 'Test Category',
  language: 'Test Language',
  isbn10: 'Test ISBN10',
  isbn13: 'Test ISBN13',
  description: 'Test Description'
};

export const mockPreloadedStateEmpty = {
  profile: {
    email: '',
    token: '',
    error: {} as Error,
    isLogged: false,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockPreloadedStateFilled = {
  profile: {
    email: 'davidson',
    token: 'teste',
    error: {} as Error,
    isLogged: true,
    isLoading: false
  },
  books: {
    isEnd: false,
    loadingFetchBooks: false,
    error: {} as Error,
    booksData: [mockBook],
    category: { key: '', title: '' },
    search: ''
  }
};

export const mockFetchDataBooks: IMockFetchDataBooks = {
  data: [
    {
      ...mockBook,
      id: '2',
      title: 'Harry Potter 1'
    },
    {
      ...mockBook,
      id: '3',
      title: 'Harry Potter 2'
    },
    {
      ...mockBook,
      id: '4',
      title: 'Harry Potter 3'
    },
    {
      ...mockBook,
      id: '5',
      title: 'Harry Potter 4'
    },
    {
      ...mockBook,
      id: '6',
      title: 'Harry Potter 5'
    }
  ],
  page: 1,
  totalPages: 1,
  totalItems: 2
};
