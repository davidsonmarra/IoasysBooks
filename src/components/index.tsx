export { Input } from './form/Input';
export { SearchInput } from './form/SearchInput';
export { FilterModal } from './FilterModal';
export { CardBook } from './CardBook';
export { ShimmerCardBook } from './ShimmerCardBook';
