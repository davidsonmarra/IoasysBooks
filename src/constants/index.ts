const constants = {
  BASE_URL: 'https://books.ioasys.com.br/api/v1',
  TIMEOUT: 8000,
  asyncStorageUserKey: '@books_asyncStorageUserKey',
  asyncStorageUserId: '@books_asyncStorageUserId',
  asyncStorageUserRefresh: '@books_asyncStorageUserRefresh'
};

export default constants;
