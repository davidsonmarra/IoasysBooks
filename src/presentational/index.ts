export { SignIn } from './SignIn';
export { BookDetails } from './BookDetails';
export { Home } from './Home';
