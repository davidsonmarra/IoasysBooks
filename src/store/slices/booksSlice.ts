import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AxiosError } from 'axios';
import BookDTO from '../../@types/BookDTO';
import { CategoryProps } from '../../@types/CategoryProps';

interface Props {
  isEnd: boolean;
  loadingFetchBooks: boolean;
  error: Error | AxiosError;
  booksData: BookDTO[];
  category: CategoryProps;
}

const initialState: Props = {
  isEnd: true,
  loadingFetchBooks: false,
  error: {} as Error | AxiosError,
  booksData: [],
  category: { key: '', title: '' }
};

export interface IFetchBooks {
  offset: number;
  category: CategoryProps;
  search: string;
}

const counterSlice = createSlice({
  name: 'books',
  initialState,
  reducers: {
    FETCH_BOOKS: (state, { payload }: PayloadAction<IFetchBooks>) => ({
      ...state,
      loadingFetchBooks: true,
      isEnd: false,
      category: payload.category,
      error: {} as Error | AxiosError
    }),
    FETCH_BOOKS_SUCCESS: (state, { payload }: PayloadAction<BookDTO[]>) => ({
      ...state,
      isEnd: payload?.length < 15,
      loadingFetchBooks: false,
      error: {} as Error | AxiosError,
      booksData: state.booksData.concat(payload)
    }),
    FETCH_BOOKS_ERROR: (state, { payload }: PayloadAction<Error | AxiosError>) => ({
      ...state,
      loadingFetchBooks: false,
      error: payload
    }),
    RESET_BOOKS: state => ({
      ...state,
      isEnd: false,
      loadingFetchBooks: false,
      error: {} as Error | AxiosError,
      booksData: [],
      category: { key: '', title: '' },
      search: ''
    })
  }
});

const { actions, reducer } = counterSlice;

export const { FETCH_BOOKS, FETCH_BOOKS_SUCCESS, FETCH_BOOKS_ERROR, RESET_BOOKS } = actions;
export default reducer;
