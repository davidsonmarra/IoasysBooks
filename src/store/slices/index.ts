import { combineReducers } from '@reduxjs/toolkit';

import profile from './profileSlice';
// import books from './booksSlice';

export default combineReducers({ profile });
