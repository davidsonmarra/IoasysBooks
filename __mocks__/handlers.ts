import { rest } from 'msw';
import { mockFetchDataBooks, mockFetchDataRefresh, mockFetchDataSignIn } from '.';
import constants from '../src/constants';

export const handlers = [
  rest.get(`${constants.BASE_URL}/books`, (req, res, ctx) => {
    const search: string = req.url.searchParams.get('title') || '';
    return search.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataBooks));
  }),
  rest.post(`${constants.BASE_URL}/auth/sign-in`, async (req, res, ctx) => {
    const { email } = await req.json();
    return email.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataSignIn));
  }),
  rest.post(`${constants.BASE_URL}/auth/refresh-token`, async (req, res, ctx) => {
    const { refreshToken } = await req.json();
    return refreshToken.includes('error')
      ? res(
          ctx.status(401),
          ctx.json({
            errorMessage: `Error 401`
          })
        )
      : res(ctx.json(mockFetchDataRefresh));
  })
];
