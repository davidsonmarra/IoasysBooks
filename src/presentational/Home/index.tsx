import React, { useEffect, useState } from 'react';
import { FlatListProps, FlatList, Keyboard, RefreshControl } from 'react-native';
import styled, { useTheme } from 'styled-components/native';
import LogoSvg from '../../assets/logo-dark.svg';
import FilterSvg from '../../assets/filter.svg';
import TitleSvg from '../../assets/title-dark.svg';
import LogoutSvg from '../../assets/logout.svg';
import { RFValue } from 'react-native-responsive-fontsize';
import BookDTO from '../../@types/BookDTO';
import { CardBook, FilterModal, SearchInput, ShimmerCardBook } from '../../components';
import { FETCH_BOOKS, RESET_BOOKS } from '../../store/slices/booksSlice';
import { useDispatch, useSelector } from 'react-redux';
import { IRootState } from '../../store/store';
import { LOGOUT } from '../../store/slices/profileSlice';
import { NavigationProp, useNavigation } from '@react-navigation/native';
import AuthRootStackParamList from '../../@types/AuthRootStackParamList';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import { schema } from './schema';

export interface IFormSearch {
  search: string;
}

export function Home() {
  const { colors } = useTheme();
  const [filterModalVisible, setFilterModalVisible] = useState(false);
  const [offset, setOffset] = useState(1);
  const { navigate } = useNavigation<NavigationProp<AuthRootStackParamList, 'Home'>>();
  const dispatch = useDispatch();
  const { isEnd, booksData, loadingFetchBooks, category } = useSelector(
    ({ books }: IRootState) => books
  );
  const { control, getValues, handleSubmit, reset } = useForm<IFormSearch>({
    resolver: yupResolver(schema),
    mode: 'onSubmit'
  });

  const handleFilterModal = () => setFilterModalVisible(prev => !prev);

  const handleLogout = () => {
    dispatch(RESET_BOOKS());
    dispatch(LOGOUT());
  };

  const onReset = () => {
    setOffset(1);
    reset();
    dispatch(RESET_BOOKS());
    dispatch(FETCH_BOOKS({ offset: 1, category, search: '' }));
  };

  const onSubmit = (form: IFormSearch) => {
    Keyboard.dismiss();
    setOffset(1);
    dispatch(RESET_BOOKS());
    dispatch(FETCH_BOOKS({ offset: 1, category, search: form.search }));
  };

  const onEndReached = () => {
    if (!isEnd) {
      dispatch(FETCH_BOOKS({ offset: offset + 1, category, search: getValues('search') || '' }));
      setOffset(offset + 1);
    }
  };

  const renderListFooterComponent = () =>
    loadingFetchBooks ? (
      <>
        <ShimmerCardBook />
        <ShimmerCardBook />
        <ShimmerCardBook />
      </>
    ) : (
      <StyledFooter />
    );

  function handleGoToBookDetails(book: BookDTO) {
    navigate('BookDetailsScreen', { book });
  }

  const renderItem = ({ item }: { item: BookDTO }) => (
    <CardBook data={item} handleGoToBookDetails={book => handleGoToBookDetails(book)} />
  );

  const refreshControl = (
    <RefreshControl
      colors={[colors.button]}
      tintColor={colors.button}
      refreshing={loadingFetchBooks}
      onRefresh={() => onSubmit({ search: getValues('search') || '' })}
    />
  );

  useEffect(() => {
    dispatch(FETCH_BOOKS({ offset, category, search: getValues('search') || '' }));
  }, []);

  return (
    <StyledContainer>
      <StyledHeader>
        <StyledTitle>
          <StyledLogo />
          <TitleSvg />
        </StyledTitle>
        <StyledLogoutButton accessibilityLabel='logout' onPress={handleLogout}>
          <LogoutSvg />
        </StyledLogoutButton>
      </StyledHeader>
      <StyledSearch>
        <SearchInput
          control={control}
          name='search'
          placeholder='Procure um livro'
          autoCorrect={false}
          autoCapitalize='none'
          searchValue={getValues('search')}
          onSubmit={handleSubmit(onSubmit)}
          reset={onReset}
        />
        <StyledFilterButton onPress={handleFilterModal}>
          <StyledFilter />
        </StyledFilterButton>
      </StyledSearch>
      <StyledList
        data={booksData}
        refreshControl={refreshControl}
        keyExtractor={item => `${item.id}`}
        refreshing={loadingFetchBooks}
        showsVerticalScrollIndicator={false}
        renderItem={renderItem}
        onEndReached={onEndReached}
        onEndReachedThreshold={0.5}
        scrollEventThrottle={16}
        ListFooterComponent={renderListFooterComponent}
        keyboardDismissMode='on-drag'
      />
      <FilterModal
        visible={filterModalVisible}
        handleModal={handleFilterModal}
        setOffset={setOffset}
        getSearch={() => getValues('search')}
      />
    </StyledContainer>
  );
}

export const StyledContainer = styled.View`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.background};
`;

export const StyledLogo = styled(LogoSvg)`
  margin-right: ${RFValue(16.6)}px;
`;

export const StyledHeader = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: ${RFValue(42)}px ${RFValue(16)}px 0;
`;

export const StyledTitle = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

export const StyledLogoutButton = styled.TouchableOpacity`
  width: ${RFValue(32)}px;
  height: ${RFValue(32)}px;
  justify-content: center;
  align-items: center;
  border: ${RFValue(1)}px solid ${({ theme }) => theme.colors.input_background};
  border-radius: ${RFValue(16)}px;
`;

export const StyledSearch = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-top: ${RFValue(32)}px;
  padding: 0 ${RFValue(16)}px ${RFValue(32)}px;
`;

export const StyledFilterButton = styled.TouchableOpacity``;

export const StyledFilter = styled(FilterSvg).attrs({
  width: RFValue(24),
  height: RFValue(24)
})``;

export const StyledList = styled(
  FlatList as new (props: FlatListProps<BookDTO>) => FlatList<BookDTO>
).attrs({
  contentContainerStyle: {
    alignItems: 'center',
    paddingBottom: RFValue(24)
  }
})``;

export const StyledFooter = styled.View`
  height: ${RFValue(100)}px;
`;
