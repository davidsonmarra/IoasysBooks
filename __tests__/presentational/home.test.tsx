import { Home } from '../../src/presentational';
import { mockBook, mockFetchDataBooks, mockPreloadedStateFilled } from '../../__mocks__';
import {
  matchSnapshotWithProviders,
  fireEvent,
  renderWithCustomProviders,
  act,
  waitFor,
  cleanup
} from '../test-utils';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { FETCH_BOOKS_SUCCESS, FETCH_BOOKS_ERROR } from '../../src/store/slices/booksSlice';

const server = setupServer(
  rest.get('https://books.ioasys.com.br/api/v1/books', (req, res, ctx) => {
    const search: string = req.url.searchParams.get('title') || '';
    return search.includes('error')
      ? res(ctx.status(401), ctx.json(mockFetchDataBooks))
      : res(
          ctx.json({
            errorMessage: `Error 401`
          })
        );
  })
);

jest.mock('redux-saga/effects', () => ({
  ...jest.requireActual('redux-saga/effects'),
  put: jest.fn()
}));

const mockedUsedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    navigate: mockedUsedNavigate
  })
}));

describe('Home', () => {
  afterEach(() => {
    cleanup();
    server.resetHandlers();
  });

  afterAll(() => server.close());

  it('should match a snapshot', () => {
    matchSnapshotWithProviders(<Home />);
  });

  it('should logout function be called correctly', () => {
    const {
      render: { getByLabelText },
      store
    } = renderWithCustomProviders(<Home />, mockPreloadedStateFilled);
    const logoutButton = getByLabelText(/logout/i);
    expect(store.getState().profile.email).toBeTruthy();

    fireEvent.press(logoutButton);

    expect(store.getState().profile.email).toBeFalsy();
  });

  it('should search for a book and clear search', async () => {
    server.listen();
    const {
      render: { getByPlaceholderText, getByTestId },
      store
    } = renderWithCustomProviders(<Home />, mockPreloadedStateFilled);
    const searchInput = getByPlaceholderText(/procure um livro/i);
    const searchSubmitButton = getByTestId(/search-submit-button/i);
    fireEvent.changeText(searchInput, 'harry potter');

    await waitFor(() => act(() => fireEvent.press(searchSubmitButton)));
    act(() => store.dispatch(FETCH_BOOKS_SUCCESS(mockFetchDataBooks.data)));

    expect(store.getState().books.booksData).toEqual(mockFetchDataBooks.data);

    await waitFor(() => {
      act(() => fireEvent.press(searchSubmitButton));
      act(() => store.dispatch(FETCH_BOOKS_SUCCESS([mockBook])));
    });

    expect(store.getState().books.booksData).toEqual([mockBook]);
  });

  it('should call onEndReached correctly', () => {
    const {
      render: { getByTestId },
      store
    } = renderWithCustomProviders(<Home />, mockPreloadedStateFilled);
    const booksFlatlist = getByTestId(/books-flatlist/i);

    fireEvent(booksFlatlist, 'onEndReached');
    act(() => store.dispatch(FETCH_BOOKS_SUCCESS([])));
    fireEvent(booksFlatlist, 'onEndReached');

    expect(store.getState().books.isEnd).toBeTruthy();
  });

  it('should navigate to book details screen', () => {
    const {
      render: { getByTestId }
    } = renderWithCustomProviders(<Home />, mockPreloadedStateFilled);
    const book = getByTestId(/book-item-on-flatlist/i);
    expect(mockedUsedNavigate).not.toHaveBeenCalled();
    fireEvent.press(book);
    expect(mockedUsedNavigate).toHaveBeenCalledWith('BookDetailsScreen', { book: mockBook });
  });

  it('should filter modal works correctly', () => {
    const {
      render: { getByLabelText, getByTestId }
    } = renderWithCustomProviders(<Home />, mockPreloadedStateFilled);
    const filterButton = getByLabelText(/filtros/i);
    const filterModal = getByTestId(/filter-modal/i);

    expect(filterModal.props.visible).toBeFalsy();
    fireEvent.press(filterButton);
    expect(filterModal.props.visible).toBeTruthy();
  });

  it('should work correctly on 401 error', async () => {
    server.listen();
    const {
      render: { getByPlaceholderText, getByTestId },
      store
    } = renderWithCustomProviders(<Home />, {
      ...mockPreloadedStateFilled,
      books: { ...mockPreloadedStateFilled.books, booksData: [{ ...mockBook, title: 'error' }] }
    });
    const searchInput = getByPlaceholderText(/procure um livro/i);
    const searchSubmitButton = getByTestId(/search-submit-button/i);
    fireEvent.changeText(searchInput, 'error');

    expect(store.getState().books.error).toStrictEqual({});
    await waitFor(() => act(() => fireEvent.press(searchSubmitButton)));
    act(() => store.dispatch(FETCH_BOOKS_ERROR(new Error('message'))));
    expect(store.getState().books.error).toStrictEqual(Error('message'));
  });
});
