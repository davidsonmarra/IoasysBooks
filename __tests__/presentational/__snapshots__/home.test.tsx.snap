// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E5E5E5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 31,
          "paddingRight": 31,
          "paddingTop": 82,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <SvgMock
        style={
          Array [
            Object {
              "marginRight": 33,
            },
          ]
        }
      />
      <SvgMock />
    </View>
    <View
      accessibilityLabel="logout"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#00000052",
          "borderRadius": 31,
          "borderStyle": "solid",
          "borderWidth": 2,
          "height": 63,
          "justifyContent": "center",
          "opacity": 1,
          "width": 63,
        }
      }
    >
      <SvgMock />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 63,
          "paddingBottom": 63,
          "paddingLeft": 31,
          "paddingRight": 31,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#00000052",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 118,
          "justifyContent": "flex-end",
          "opacity": 1,
          "paddingBottom": 0,
          "paddingLeft": 22,
          "paddingRight": 31,
          "paddingTop": 0,
          "width": "85%",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Procure um livro"
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "color": "#333333",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Heebo-Regular",
              "fontSize": 27,
              "height": "100%",
            },
          ]
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "opacity": 1,
          }
        }
        testID="search-submit-button"
      >
        <SvgMock
          height={47}
          style={
            Array [
              Object {},
            ]
          }
          width={47}
        />
      </View>
    </View>
    <View
      accessibilityLabel="filtros"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <SvgMock
        height={47}
        style={
          Array [
            Object {},
          ]
        }
        width={47}
      />
    </View>
  </View>
  <RCTScrollView
    ListFooterComponent={[Function]}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "paddingBottom": 47,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    keyboardDismissMode="on-drag"
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        colors={
          Array [
            "#B22E6F",
          ]
        }
        onRefresh={[Function]}
        refreshing={false}
        tintColor="#B22E6F"
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {},
      ]
    }
    testID="books-flatlist"
    viewabilityConfigCallbackPairs={Array []}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Array [
              Object {
                "height": 196,
              },
            ]
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000052",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="filter-modal"
    transparent={true}
    visible={false}
  />
</View>
`;
